keps/sig-api-machinery/2839-in-use-protection/README.md

```diff
@@ -0,0 +1,766 @@
+
+#### Unresolved issues
+
+- Decision: When you delete a namespace and an object in that namespace has a lien:
+  - Does the namespace and object get deleted anyway?
+    - [x] Yes.
+    - [ ] No. See below cases:
+      - Does the namespace deletion get blocked (i.e., the delete request fails) if any object in the namespace has a lien?
+        - No.
+      - Does the other objects in the namespace get deleted, but that not object, preventing the namespace deletion from completing?
+        - [x] No. Namespace lifecycle cleanup will not honor liens. if you delete the namespace, the content will be removed.
+	    - [ ] Yes. See below case:
+          - Does the namespace deletion not proceed until all objects in the namespace are free of liens?
+            - [x] No. Namespace lifecycle cleanup will not honor liens. If you delete the namespace, the content will be removed.
+	        - [ ] ~~Yes. Namespace lifecycle cleanup will honor liens. If you delete the namespace, it won't be deleted until all the content will be deleted.~~
+	        - [ ] Partially yes. Namespace lifecycle cleanup will honor liens. If you delete the namespace, it won't start deleting the content with liens until all the content without liens will be deleted (Current KEP's description (Also see [Namespace Deletion](#namespace-deletion))).
+- Name:
+    - [x] "liens" is short and precise, but an unusual word
+    - [ ] Other candidates: InUse, deleteInhibitors, hold, lease, claim, deletionBlockers, protections, guards
+- Subresource:
+  - Do you need a special permission to add a lien?
+    - [x] Yes. Liens can only be changed via a call to resource/liens subresource. Create/update permission for the resource/liens subresource is required.
+	- [ ] ~~No. Only create/update permission forthe resource is required.~~
+	- [ ] Not always. Create/Update permission for resource or update permission for resource/liens subresource are required (Current KEP's description (Also see [Race of removing/adding liens](#race-of-removingadding-liens))).
+  - To remove a lien?
+    - [x] Delete permission for the resource/liens subresource is required.
+	- [ ] ~~Delete permission for the resource is required.~~
+	- [ ] Delete permission for the resource or delete permission for the resource/liens subresource are required (Current KEP's description).
+  - To delete anyway and ignore liens?
+    - [x] `--not-block-on-lien` kubectl option will pass `NotBlockOnLien` API option in `DeleteOptions` to force delete.
+  - Who can ignore liens and how it can be specified?
+    - [x] Users who have delete permission for the resource/liens subresource without any additional configurations (If Liens can only be changed via a call to resource/liens subresource, normal users cannot use it to create undeletable resources).
+    - [ ] Users who are specified by kube-apiserver's command line options, `--allowed-not-block-on-lien-users` and `--allowed-not-block-on-lien-groups` (Current KEP's description).
```

Comment:
If we add these options, are we baking in the decisions that exemptions are always granted to users (and not groups, or via extra auth data)?